juriscraper/opinions/united_states/federal_appellate/ca10.py

```diff
@@ -24,22 +38,43 @@ def _get_download_urls(self):
     def _get_case_dates(self):
         dates = []
         for date_string in self.html.xpath("//item/pubdate/text()"):
-            date_only = " ".join(date_string.split(" ")[1:4])
-            dates.append(
-                date.fromtimestamp(
-                    time.mktime(time.strptime(date_only, "%d %b %Y"))
+            try:
+                span_element = html.etree.fromstring(str(date_string))
+                date_string = span_element.xpath("/span/text()")[0]
+                date_only = " ".join(date_string.split(" ")[1:4])
+                dates.append(
+                    date.fromtimestamp(
+                        time.mktime(time.strptime(date_only, "%b %d %Y"))
+                    )
+                )
+            except:
+                logger.error(f"Error while parsing case date: {date_string}")
+                raise InsanityException(
+                    f"Error while parsing case date: {date_string}"
                 )
-            )
         return dates
 
     def _get_docket_numbers(self):
         return [
-            e.split(":")[1]
-            for e in self.html.xpath("//item/description/text()[1]")
+            e.split(" - ")[0].split(":")[1]
```

Comment:
It'd be helpful to have a comment here showing what the text usually looks like before the splitting happens.

Reply:
Added